extensions/amp-story/1.0/pagination-buttons.js

```diff
@@ -244,28 +229,25 @@ export class PaginationButtons {
     }
   }
 
+  /**
+   * Updates element based on visibility state.
+   * @param {?PaginationButton} paginationButton
+   * @param {boolean} isVisible
+   * @private
+   */
+  toggleEnabled_(paginationButton, isVisible) {
+    const {buttonElement, element} = paginationButton;
+    element?.classList.toggle('i-amphtml-story-button-hidden', !isVisible);
+    buttonElement?.toggleAttribute('disabled', !isVisible);
+  }
+
   /**
    * Reacts to system UI visibility state updates.
    * @param {boolean} isVisible
    * @private
    */
   onSystemUiIsVisibleStateUpdate_(isVisible) {
-    if (isVisible) {
-      this.backButton_.updateState(
-        /** @type {!PaginationButtonStateDef} */ (
-          devAssert(this.backButtonStateToRestore_)
-        )
-      );
-      this.forwardButton_.updateState(
-        /** @type {!PaginationButtonStateDef} */ (
-          devAssert(this.forwardButtonStateToRestore_)
-        )
-      );
-    } else {
-      this.backButtonStateToRestore_ = this.backButton_.getState();
-      this.backButton_.updateState(BackButtonStates.HIDDEN);
-      this.forwardButtonStateToRestore_ = this.forwardButton_.getState();
-      this.forwardButton_.updateState(ForwardButtonStates.HIDDEN);
-    }
+    this.toggleEnabled_(this.backButton_, isVisible);
+    this.toggleEnabled_(this.forwardButton_, isVisible);
```

Comment:
`toggleEnabled_` will set the visibility state of these two buttons to the opposite of `isVisible`. Is that intentional?

Reply:
It's intentional. It sets hidden states based on `isVisible`.